src/cli/input.rs

```diff
@@ -63,7 +66,16 @@ pub fn parse_input(args: &ArgMatches) -> Result<Input> {
         return Ok(Input::GenerateFlamegraph);
     }
 
-    let input_string = args.value_of(ARGUMENT_VALUE_NAME_INPUT).unwrap_or(""); // No input is provided, fallback to an empty string
+    let input_string = &match args.value_of(ARGUMENT_VALUE_NAME_INPUT) {
+        Some(v) => v.to_string(),
+        None => {
+            if args.is_present(FLAG_NAME_TUI_MODE) {
+                String::new() // Fallback to an empty string
+            } else {
+                return Err(Box::new(Error::MissingInput));
```

Comment:
This is just my opinion and might be a bit nitpicking, I think since we are already doing pattern matching, maybe we could do it this way:
```
let input_string = match args.value_of(ARGUMENT_VALUE_NAME_INPUT) {
        Some(v) => Ok(v),
        None => {
            if args.is_present(FLAG_NAME_TUI_MODE) {
                Ok("") // Fallback to an empty string
            } else {
                Err(Box::new(Error::MissingInput))
            }
        }
}?;
```

Here we make the `match` expression return a `Result<&str>`, and unwrap it at the end of the expression (or, if the result is an error, we bail and bubble up the error).

Not 100% sure whether `&str` would work, if it does then great (we saved a copy of the string), if it doesn't, then what you have is fine, since this function only runs once per launch.

Reply:
I think this is a good suggestion. I applied these changes in 2556945